src/requests/parameter_wrappers.rs

```diff
@@ -1,235 +1,188 @@
+            ))?
+        };
+    }
+    let value: CqlValue = match typ.base_type {
+        CqlType::Ascii => CqlValue::Ascii(get_element!(String)),
+        CqlType::Boolean => CqlValue::Boolean(get_element!(bool)),
+        CqlType::Blob => CqlValue::Blob(get_element!(Buffer).to_vec()),
+        CqlType::Counter => CqlValue::Counter(Counter(bigint_to_i64(
+            get_element!(BigInt),
+            "Value cast into counter type shouldn't overflow i64",
+        )?)),
+        CqlType::Decimal => todo!(),
+        CqlType::Date => CqlValue::Date(get_element!(&LocalDateWrapper).get_cql_date()),
+        CqlType::Double => CqlValue::Double(get_element!(f64)),
+        CqlType::Duration => CqlValue::Duration(get_element!(&DurationWrapper).get_cql_duration()),
+        CqlType::Float => CqlValue::Float(get_element!(f64) as f32),
+        CqlType::Int => CqlValue::Int(get_element!(i32)),
+        CqlType::BigInt => CqlValue::BigInt(bigint_to_i64(
+            get_element!(BigInt),
+            "Cannot fit value in CqlBigInt",
+        )?),
+        CqlType::Text => CqlValue::Text(get_element!(String)),
+        CqlType::Timestamp => CqlValue::Timestamp(CqlTimestamp(bigint_to_i64(
+            get_element!(BigInt),
+            "timestamp cannot overflow i64",
+        )?)),
+        CqlType::Inet => CqlValue::Inet(get_element!(&InetAddressWrapper).get_ip_addr()),
+        CqlType::List => CqlValue::List(cql_value_vec_from_array(
+            typ.support_type_1
+                .as_ref()
+                .expect("Expected support type for list"),
+            &get_element!(Array),
+        )?),
+        CqlType::Map => CqlValue::Map(cql_value_vec_from_map(typ, &get_element!(Array))?),
+        CqlType::Set => CqlValue::Set(cql_value_vec_from_array(
+            typ.support_type_1
+                .as_ref()
+                .expect("Expected support type for list"),
+            &get_element!(Array),
+        )?),
+        CqlType::UserDefinedType => todo!(),
+        CqlType::SmallInt => CqlValue::SmallInt(
+            get_element!(i32)
+                .try_into()
+                .map_err(|_| js_error("Value must fit in i16 type to be small int"))?,
+        ),
+        CqlType::TinyInt => CqlValue::TinyInt(
+            get_element!(i32)
+                .try_into()
+                .map_err(|_| js_error("Value must fit in i8 type to be tiny int"))?,
+        ),
+        CqlType::Time => CqlValue::Time(get_element!(&LocalTimeWrapper).get_cql_time()),
+        CqlType::Timeuuid => CqlValue::Timeuuid(CqlTimeuuid::from_bytes(
+            get_element!(&UuidWrapper).get_cql_uuid().into_bytes(),
+        )),
+        CqlType::Tuple => CqlValue::Tuple(cql_value_vec_from_tuple(typ, &get_element!(Array))?),
+        CqlType::Uuid => CqlValue::Uuid(get_element!(&UuidWrapper).get_cql_uuid()),
+        CqlType::Varint => todo!(),
+        CqlType::Custom => return Err(js_error("Improper type information")),
```

Comment:
:wrench: The error message is not helpful. Please say explicitly that `CqlType::Custom` was passed.